### PR TITLE
Remove historical feature in serving store

### PR DESCRIPTION
**Pull request by pradithya**
Remove historical value of feature in serving store. This is incomplete solution and currently is blocked by the issue with late data overwriting newer data.

It contains breaking changes in serving and all the client will have to be updated

/hold

### CODE REVIEW

**Comment by pradithya** [comment]
It's ready for review, let me know if you are against the new API structure
/assign @tims 
/assign @zhilingc

File: protos/feast/serving/Serving.proto [hunk 1]
```diff
@@ -26,51 +26,29 @@ option java_outer_classname = "ServingAPIProto";
...
-    // LAST : return only the latest value based on timestamp.
-    // LIST : return list of historical data sorted by timestamp.
-    RequestType type = 2;
-    // only applicable to LIST.
-    // length of the returned list <= limit.
-    // default = 0
-    int32 limit = 3;
+    repeated string featureId = 3;
+    // [optional] time range filter. If not specified all feature with any timestamp will be returned.
+    TimestampRange timeRange = 4;
```
Comment (tims) [inline protos/feast/serving/Serving.proto]: This seems a bit misleading. We should remove time range. 

We return the associated timestamps, so we can leave it to the client to decide if they care things are too old.

File: serving/src/main/java/feast/serving/service/BigTableFeatureStorage.java [hunk 2]
```diff
@@ -168,84 +123,35 @@ public BigTableFeatureStorage(Connection connection) {
...
+
     List<Get> gets = new ArrayList<>();
     for (String entityId : entityIds) {
       String entityIdPrefix = DigestUtils.sha1Hex(entityId.getBytes()).substring(0, 7);
       BigTableRowKey btKey = createRowKey(entityIdPrefix, entityId, "0");
       Get get = new Get(btKey.toByteArray());
       get.addColumn(columnFamily, featureIdBytes);
       try {
-        // for some reason Get.readVersions has checked exception.
+        if (!featureSpec.getGranularity().equals(Enum.NONE) && tsRange != null) {
```
Comment (tims) [inline serving/src/main/java/feast/serving/service/BigTableFeatureStorage.java]: We should not need this.

File: serving/src/main/java/feast/serving/service/RedisFeatureStorage.java [hunk 3]
```diff
@@ -79,145 +68,44 @@ public RedisFeatureStorage(JedisPool jedisPool, Tracer tracer) {
...
-            count++;
-            featureCountMap.put(bucket.getFeatureId(), count);
-            entityCountMap.put(bucket.getEntityId(), featureCountMap);
-          } else {
-            break;
           }
         }
       }
+      scope.span().log("completed request creation");
+      List<FeatureValue> unfilteredResult = sendAndProcessMultiGet(getRequests);
```
Comment (tims) [inline serving/src/main/java/feast/serving/service/RedisFeatureStorage.java]: isn't all the data stored in granularity None now?

Comment (tims) [reply serving/src/main/java/feast/serving/service/RedisFeatureStorage.java]: so we can cut some of this?

Comment (tims) [reply serving/src/main/java/feast/serving/service/RedisFeatureStorage.java]: To be precise, the only thing being written has a bucketId of 0L.

File: serving/src/main/java/feast/serving/service/BigTableFeatureStorage.java (hunk 2 above)
Comment (tims) [reply serving/src/main/java/feast/serving/service/BigTableFeatureStorage.java]: Similarly to Redis, the we now only write to bigtable with a reversedMillis in the row key of "0", effectively always granularity of None, so there is no need to consider granularity at all.

File: protos/feast/serving/Serving.proto (hunk 1 above)
Comment (pradithya) [reply protos/feast/serving/Serving.proto]: I agree, i'll amend the PR

**Comment by pradithya** [comment]
@tims I updated the PR to exclude server-side timestamp filtering. The timestamp filter is now done in client side.

**Comment by tims** [comment]
Last feedback, ts_range in the sdk should be DateTime objects not strings.

**Comment by pradithya** [comment]
@zhilingc can you please review

**Pull request merged**